borg/helpers.py

```diff
@@ -259,7 +259,7 @@ def update_excludes(args):
 
 def adjust_patterns(paths, excludes):
     if paths:
-        return (excludes or []) + [IncludePattern(path) for path in paths] + [ExcludePattern('*')]
+        return (excludes or []) + [PathPrefixPattern(path) for path in paths] + [FnmatchPattern('*')]
```

Comment:
> why can't we have any mechanism as well for including as for excluding?

We can. I intend to move this code into a class together with `exclude_path` so that it's decoupled from excluding. The interface will be something along these lines:

```
class PatternMatcher:
  def __init__(self, default_style): …
  def add_include(self, pattern): …
  def add_exclude(self, pattern): …
  def match(self, path): …
```

Each added pattern is stored into a single, ordered list together with a flag for include/exclude. `create` will use a default pattern style of `fm` (fnmatch) as is, and `extract` will use the path prefix style by default (still to be added to `parse_pattern`).

I'm sorry not having made this clear before. I mainly wanted see whether the refactored tests still work on Mac OS X.